pkg/R/sparkR.R

```diff
@@ -121,45 +119,38 @@ sparkR.init <- function(
   if (sparkRExistingPort != "") {
     sparkRBackendPort <- sparkRExistingPort
   } else {
+    # TODO: test the random port and retry, or use httpuv:::startServer
+    callbackPort = sample(40000, 1) + 10000
```

Comment:
We pick the port first, and pass it into JVM, then we can not try and change the port anymore.

The final solution maybe use httpuv to launch a http server in the background with random port, then pass the real port into JVM, JVM send back the port of backend via HTTP request.